core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/CarbonColumnVectorImpl.java

```diff
@@ -70,10 +72,31 @@
 
   private CarbonColumnVector dictionaryVector;
 
+  private CarbonColumnVectorImpl childrenVector;
+
   private LazyPageLoader lazyPage;
 
   private boolean loaded;
 
+  public boolean isComplex() {
+    return false;
```

Comment:
please check if can add

public DataType getDataType() {
    return dataType;
  }

and check it is complex or not